tsk/fs/hfs.c

```diff
@@ -2675,7 +2675,7 @@ hfs_attr_walk_special(const TSK_FS_ATTR * fs_attr,
     }
 
     // Allocate two buffers of the compression unit size.
-    rawBuf = (char *) tsk_malloc(COMPRESSION_UNIT_SIZE);
+    rawBuf = (char *) tsk_malloc(COMPRESSION_UNIT_SIZE + 1);
     uncBuf = (char *) tsk_malloc(COMPRESSION_UNIT_SIZE);
```

Comment:
Does len end up being COMPRESSION_UNIT_SIZE+1 if the unit is not compressed and the new bounds checks prevent it from processing those?

The other minor change I'll make to this code after it is merged (or you can do it if you want) would be to allocate and free uncBuf and rawBuf closer to the loop that they are used it.  THey seem to be allocated early and freed late, which increases the number of places that we need to have error handling to free them.